includes/Jobs/GenerateProductFeed.php

```diff
@@ -18,18 +19,84 @@ class GenerateProductFeed extends AbstractChainedJob {
 
 	use BatchQueryOffset, LoggingTrait;
 
+	/**
+	 * Array of processing results.
+	 *
+	 * @var array $processed_items.
+	 */
+	protected $processed_items = array();
+
+	/**
+	 * Feed file creation and manipulation utility.
+	 *
+	 * @var FacebookFeedFileHandler $feed_file_handler.
+	 */
+	protected $feed_file_handler;
+
+	/**
+	 * Feed file creation and manipulation utility.
+	 *
+	 * @var FeedDataExporter $feed_data_exporter.
+	 */
+	protected $feed_data_exporter;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param ActionSchedulerInterface $action_scheduler   Action Scheduler facade.
+	 * @param FacebookFeedFileHandler  $feed_file_handler  Feed file creation and manipulation handler.
+	 * @param FeedDataExporter         $feed_data_exporter Handling of file data.
+	 */
+	public function __construct( ActionSchedulerInterface $action_scheduler, $feed_file_handler, $feed_data_exporter ) {
```

Comment:
@danielbitzer This is a part of the AS framework. Is this something we need to look into?

Reply:
From what I see this type of type hinting is available since PHP5.0: https://mlocati.github.io/articles/php-type-hinting.html

Since this is interacting with a library that was built in a certain way I would leave it as it is.

Reply:
I think we should be type-hinting as much as possible in any newly written code. PHP 7.0 supports most type-hints and function return types. We can possible discuss this in our meeting since it affects the code base more broadly than this PR.

Reply:
I agree - I don't think we need to push hard, but should be consistent within each function - all args are type hinted.